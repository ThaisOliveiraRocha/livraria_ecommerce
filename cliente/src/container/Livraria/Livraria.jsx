import React, { useState, useEffect } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import Layout from "../../components/Layout";
import Card from "./components/Card/Card";
import ModalComponent from "../../components/ModalComponent";
import { removeLivro, showModal } from "../../store/actions";
import { withRouter } from "react-router-dom";
import Slider from "../../components/Slider";
import {
  LivrariaContainer,
  SliderContainer,
  CardContainer,
  ProductsContainer,
} from "./Livraria.styles";
import Titulo from "../../components/Title";
import Texto from "../../components/Text";

import { makeStyles } from "@material-ui/core/styles";
import InputBase from "@material-ui/core/InputBase";
import Divider from "@material-ui/core/Divider";
import IconButton from "@material-ui/core/IconButton";
import SearchIcon from "@material-ui/icons/Search";
import Button from "../../components/Button";
import KeyboardArrowDownIcon from "@material-ui/icons/KeyboardArrowDown";
import { replaceSpecialCharacter } from "../../utils";

const useStyles = makeStyles((theme) => ({
  root: {
    padding: "2px 4px",
    display: "flex",
    alignItems: "center",
    width: "50%",
    border: "1px solid #2a7886",
    borderRadius: "3px",
    backgroundColor: "white",
    marginTop: "30px",
    marginBottom: "20px",
  },
  input: {
    marginLeft: theme.spacing(1),
    flex: 1,
  },
  iconButton: {
    padding: 10,
  },
  divider: {
    height: 24,
    margin: 4,
  },
}));

const Livraria = ({ livros, showModal, funcaoConfirma, cancel }) => {
  const classes = useStyles();
  const [bookList, setBookList] = useState([]);

  useEffect(() => {
    setBookList(livros);
  }, [livros]);

  const filterOptions = (value) => {
    console.log("value ", value);
    let novosLivros = livros.filter(
      (item) =>
      replaceSpecialCharacter(item.titulo).indexOf(replaceSpecialCharacter(value)) === 0
    );
    setBookList(novosLivros);
  };

  return (
    <Layout>
      {showModal && (
        <ModalComponent
          temaModal="Deseja excluir o item?"
          mensagem="O item selecionado será excluído permanentemente."
          onClose={cancel}
          funcaoConfirma={funcaoConfirma}
          cancel={cancel}
        />
      )}
      <LivrariaContainer>
        <SliderContainer>
          <Slider />
        </SliderContainer>
        <ProductsContainer>
          <Titulo font="22px">Conheça nossos produtos</Titulo>
          <div className={classes.root}>
            <Button background="transparent" width="80px">
              <Texto color="#2a7886">Filtrar</Texto>
              <Texto color="#2a7886">
                <KeyboardArrowDownIcon />
              </Texto>
            </Button>
            <InputBase
              className={classes.input}
              placeholder="Pesquisar..."
              name="pesquisar"
              inputProps={{ "aria-label": "search" }}
              onChange={(e) => filterOptions(e.target.value)}
            />
            <IconButton
              type="submit"
              className={classes.iconButton}
              aria-label="search"
            >
              <SearchIcon />
            </IconButton>
          </div>

          <CardContainer>
            {bookList.map((livro, index) => {
              return <Card key={index} livro={livro} />;
            })}
          </CardContainer>
        </ProductsContainer>
      </LivrariaContainer>
    </Layout>
  );
};

Livraria.propTypes = {
  // bla: PropTypes.string,
};

const mapStateToProps = (state) => ({
  livros: state.livraria.vetLivros,
  showModal: state.livraria.removeModal,
});

const mapDispatchToProps = (dispatch) => ({
  funcaoConfirma: () => {
    dispatch(removeLivro());
  },
  cancel: () => {
    dispatch(showModal("", false));
  },
});

export default withRouter(
  connect(mapStateToProps, mapDispatchToProps)(Livraria)
);
