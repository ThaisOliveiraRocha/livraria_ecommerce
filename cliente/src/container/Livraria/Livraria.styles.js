import styled from "styled-components";

export const LivrariaContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  width: 100%;
  padding: 0 20px 0 20px;
`;

export const SliderContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  margin-bottom: 40px;
  position: relative;
  margin-top: 50px;
`;

export const CardContainer = styled.div`
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  justify-content: center;
  width: 100%;
`;

export const ProductsContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  width: 100%;
  height: 100%;
`;
