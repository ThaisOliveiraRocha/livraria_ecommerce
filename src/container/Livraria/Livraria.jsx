import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import Layout from "../../components/Layout";
import Card from "./components/Card/Card";
import ModalComponent from "../../components/ModalComponent";
import { removeLivro, showModal } from "../../store/actions";
import { withRouter } from "react-router-dom";

const Livraria = ({ livros, showModal, funcaoConfirma, cancel }) => (
  <Layout>
    {showModal && (
      <ModalComponent
        temaModal="Deseja excluir o item?"
        mensagem="O item selecionado será excluído permanentemente."
        onClose={cancel}
        funcaoConfirma={funcaoConfirma}
        cancel={cancel}
      />
    )}

    {livros.map((livro, index) => {
      return <Card key={index} livro={livro} />;
    })}
  </Layout>
);

Livraria.propTypes = {
  // bla: PropTypes.string,
};

const mapStateToProps = state => ({
  livros: state.livraria.livros,
  showModal: state.livraria.removeModal
});

const mapDispatchToProps = dispatch => ({
  funcaoConfirma: () => {
    dispatch(removeLivro());
  },
  cancel: () => {
    dispatch(showModal("", false));
  }
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Livraria));
