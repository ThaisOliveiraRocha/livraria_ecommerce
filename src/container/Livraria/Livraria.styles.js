import styled from "styled-components";

export const ModalBody = styled.div`
  display: flex;
  align-items: center;
  flex-direction: column;

  width: 20%;
  height: 150px;

  overflow-y: scroll; 
`;

export const Table = styled.table`
  font-size: 16px;
  font-weight: 600;
`;

export const Titulo = styled.div`
  font-size: 16px;
  font-weight: 600;
`;

export const Tr = styled.tr`
  /* display: flex;
  align-items: center;
  width: 100%; */
`;

export const Td = styled.td`
    width: 70px;
    font-weight: normal;
`;

export const ImageBook = styled.img`
  width: 40px;
  height: auto;

  margin: 10px;
`;

export const InfoBook = styled.span`
  font-size: 14px;
  margin-right: 5px;
`;

export const Input = styled.input`
  font-size: 14px;
  width: 50px;
  margin-right: 5px;
`;

export const ImageTrash = styled.img`
  width: 20px;
  height: auto;
  cursor: pointer;

  &:active {
    transform: scale(0.9, 0.9);
  }
`;

export const ButtonGroup = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
`;