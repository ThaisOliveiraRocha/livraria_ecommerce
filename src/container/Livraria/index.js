import Livraria from './Livraria';

export default Livraria;
