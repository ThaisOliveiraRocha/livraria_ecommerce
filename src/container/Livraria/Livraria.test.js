import React from "react";
import { shallow } from "enzyme";
import Livraria from "./Livraria";

describe("<Livraria />", () => {
  it("should render without crashing", () => {
    shallow(<Livraria />);
  });
  it("should render basic infos", () => {
    const wrapper = shallow(<Livraria />);
    expect(wrapper).toMatchSnapshot();
  });
  
});
