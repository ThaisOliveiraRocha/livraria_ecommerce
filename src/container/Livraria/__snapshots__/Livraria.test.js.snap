// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Livraria /> should render basic infos 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
